torch/distributions/constraints.py

```diff
@@ -297,6 +301,49 @@ def check(self, value):
         return (value == value).all()  # False for NANs.
 
 
+class _Cat(Constraint):
+    """
+    Constraint functor that applies a sequence of constraints
+    `cseq` at the submatrices at dimension `dim`,
+    each of size `lengths[dim]`, in a way compatible with :func:`torch.cat`.
+    """
```

Comment:
nit: it might be nice to show code examples of how to use these cool new transforms.

Reply:
I'd also recommend `make -C docs html` and checking that the docs look ok